cmd/juju/application/deploy.go

```diff
@@ -260,7 +265,10 @@ func NewDeployCommandForTest(newAPIRoot func() (DeployAPI, error), steps []Deplo
 }
 
 // NewDeployCommand returns a command to deploy applications.
-func NewDeployCommand() modelcmd.ModelCommand {
+func NewDeployCommand(
```

Comment:
We don't want to use any arg here and have to update all the call sites to NewDeployCommand(nil. nil)
The default values are nil and the ForTest() variant passes in non-nil values